iam/hooks.py

```diff
@@ -285,6 +288,36 @@ def is_unverified_user(cog, obj, *func_args, **func_kwargs):
         return False, "You are already verified."
     return True, None
 
+def is_strictly_verified_user(cog, obj, *func_args, **func_kwargs):
+    """Checks that user that invoked function is verified in database.
+
+    Associated cog must have bot and db as instance variables.
+
+    Args:
+        cog: Cog associated with function invocation.
+        obj: Object associated with function invocation.
+        func_args: Args supplied to function call.
+        func_kwargs: Kwargs supplied to function call.
+
+    Returns:
+        1. Boolean result of check.
+        2. Error message to supply, if check failed.
+    """
+    if not isinstance(obj, Member):
+        obj = obj.author
+    err_msg = "You must be verified to do that."
+    member = get_member(cog.bot, obj)
+    if member is not None and VER_ROLE in get_role_ids(member):
```

Comment:
sidenode: "VER_ROLE" is kinda  a bad variable name. Verified role? Version role?